docs/development/workflow/development_workflow.rst

```diff
@@ -565,8 +560,8 @@ start make a branch to serve as a backup copy of your work::
 
     git branch tmp my-new-feature # make temporary branch--will be deleted later
 
-After altering the history, e.g. with ``git rebase``, a normal ``git push``
-is prevented, and a ``git push --force`` will be required.
```

Comment:
What about [--force-with-lease](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-force-with-lease)?

Reply:
Isn't it better to specify which branch explicitly?

Reply:
I don't understand your question. The difference between `--force` and `--force-with-lease` is that `--force-with-lease` only succeeds if the remote branch is where your local repository thinks it is, so it is safer than `--force`. It has nothing to do with specifying the branch name.

Reply:
Personally, I would find `--force-with-lease` much more to type and realistically won't use it since I always give a remote and branch name. But for the sake of this doc that I don't usually read, sure.

Reply:
The difference between `--force` and `--force-with-lease` is that `--force-with-lease` fails if the branch in the remote repository has been updated since the local repository last fetched data from there, but with `--force` it is possible that the user removes commits in the remote repository that they didn't even know existed. `--force-with-lease` is therefore safer, and turning `git push --force` to `git push --force-with-lease` only takes three extra keystrokes (`-w<TAB>`).

It has nothing to do with specifying the remote and branch name explicitly.